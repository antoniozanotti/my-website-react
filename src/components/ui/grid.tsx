import { ReactNode } from "react";

export interface GridProps extends React.ComponentProps<"div"> {
  children?: ReactNode;
}

export default function Grid({ children, ...props }: GridProps) {
  let divClasses = "sm:grid sm:grid-cols-2 sm:gap-x-8 lg:gap-x-16";
  props.className = props.className
    ? divClasses + " " + props.className
    : divClasses;
  return <div {...props}>{children}</div>;
}
