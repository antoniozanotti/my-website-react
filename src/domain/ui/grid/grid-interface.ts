import { ColsType } from "./types/cols-type";

export interface GridInterface extends React.ComponentProps<"div"> {
  cols?: ColsType;
}