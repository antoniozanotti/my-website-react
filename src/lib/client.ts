import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
