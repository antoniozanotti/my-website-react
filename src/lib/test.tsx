import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
