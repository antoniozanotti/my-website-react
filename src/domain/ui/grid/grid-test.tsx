import { test, expect } from '@playwright/experimental-ct-react';
import { Grid } from "./grid";

test.describe("Grid", () => {
  test("should render div", async ({ mount }) => {
    const component = await mount(<Grid />);
    await expect(component.locator(":scope:is(div)")).toHaveCount(1);
  });

  test("should accept className", async ({ mount }) => {
    const component = await mount(<Grid className='test' />);
    await expect(component).toHaveClass(/test$/);
  });

  test("should render children", async ({ mount }) => {
    const component = await mount(<Grid>Test</Grid>);
    await expect(component).toContainText("Test");
  });

  test("should render 2 columns", async ({ mount }) => {
    const component = await mount(<Grid cols={2} />);
    await expect(component).toHaveClass(/grid-cols-2/);
  });

  test("should render 3 columns", async ({ mount }) => {
    const component = await mount(<Grid cols={3} />);
    await expect(component).toHaveClass(/grid-cols-3/);
  });
});
