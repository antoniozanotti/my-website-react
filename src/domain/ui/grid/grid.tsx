import { GridInterface } from "./grid-interface";

export function Grid({
  cols = 2,
  children,
  ...props
}: GridInterface) {
  const baseStyles = "sm:grid sm:gap-x-8 lg:gap-x-16";
  const colsStyles = cols == 3 ? "sm:grid-cols-3" : "sm:grid-cols-2";
  const divStyles = `${baseStyles} ${colsStyles}`;
  props.className = props.className
    ? `${divStyles} ${props.className}`
    : divStyles;

  return <div {...props}>{children}</div>;
}
