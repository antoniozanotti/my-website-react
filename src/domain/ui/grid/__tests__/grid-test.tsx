import { screen, render, firstComponent } from "@/lib/test";
import { Grid } from "../grid";

describe("Grid", () => {
  it("should render div", async () => {
    const { container } = render(<Grid />);
    const component = firstComponent(screen);

    expect(component).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it("should accept className", async () => {
    const { container } = render(<Grid className='test' />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/test$/);
    expect(container).toMatchSnapshot();
  });

  it("should render children", async () => {
    const { container } = render(<Grid>Test</Grid>);
    const component = firstComponent(screen);

    expect(component).toHaveTextContent("Test");
    expect(container).toMatchSnapshot();
  });

  it("should render 2 columns", async () => {
    const { container } = render(<Grid cols={2} />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/grid-cols-2/);
    expect(container).toMatchSnapshot();
  });

  it("should render 3 columns", async () => {
    const { container } = render(<Grid cols={3} />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/grid-cols-3/);
    expect(container).toMatchSnapshot();
  });
});
