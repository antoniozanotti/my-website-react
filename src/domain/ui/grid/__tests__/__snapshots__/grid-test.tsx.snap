// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should accept className 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-2 test"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid should render 2 columns 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-2"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid should render 3 columns 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid should render children 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-2"
        >
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid should render div 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-2"
        />
      </div>
    </div>
  </div>
</div>
`;
